### PR TITLE
downloadFile and Import/Export

### PR DESCRIPTION
**Pull request by ekir**
Hi!

I have unified downloadWebFile with downloadFile now. Since the original downloadFile did not work in Firefox offline, but downloadWebFile did(and also worked in other browsers like Chrome), I have replaced the content of downloadFile with downloadWebFiles content, and removed downloadWebFile so there is only one function for this purpose

I have also made Import/Export gone when you run offline since you have Load and Save

I have also made export SVG and CSV work with downloadFile, and tried them in Firefox and Chrome and they work well in both

### CODE REVIEW

File: js/API/API.js [hunk 1]
```diff
@@ -227,16 +227,11 @@ data - The data to download.
...
-	var blob = new Blob([data], {type: "octet/stream"}),
-	    url = window.URL.createObjectURL(blob);
-	a.href = url;
-	a.download = fileName;
-	a.click();
-	window.URL.revokeObjectURL(url);
-	a.remove();
+	var downloadlink = document.body.appendChild(document.createElement("a"));
+	downloadlink.download = fileName;
+	downloadlink.href = "data:text/plain;base64," + btoa(unescape(encodeURIComponent(data)));
```
Comment (scottfr) [inline js/API/API.js]: I think there are some issues with size constraints on data uri's. Maybe this is no longer an issue with modern browsers but please look into it.

File: js/Utilities.js [hunk 2]
```diff
@@ -1341,9 +1335,5 @@ var downloadButton = function(name){
 
 				var xml = (mxUtils.getXml(root));
 
-				new mxXmlRequest(builder_path + "/download.php", $.param({
-					name: "Insight Maker Diagram",
-					"format": "svg",
-					"data": xml
-				})).simulate(document, "_blank");
+				downloadFile("Insight Maker Diagram.svg", xml)
```
Comment (scottfr) [inline js/Utilities.js]: Please check this this works when serving in the web browser and it's not blocked by an anti-spam security mechanism.

Also semi-colon at end of line.

File: js/API/API.js (hunk 1 above)
Comment (ekir) [inline js/API/API.js]: I will look into that!

File: js/Utilities.js (hunk 2 above)
Comment (ekir) [inline js/Utilities.js]: I tried it in Google Chrome and Firefox without it being blocked. I can do some more tests

The semi-colon is resolved! :)

File: js/API/API.js (hunk 1 above)
Comment (ekir) [inline js/API/API.js]: After some testing my conclusion is that data-uris works best in Firefox but Blobs works best in Chrome

Firefox could do about 30 megabyte file downloads with data-uris, while Chrome stopped accepting data-uris after about 1 megabyte.

I will come back with a patch that does URI-downloads for Firefox but Blob downloads for Chrome

Comment (ekir) [inline js/API/API.js]: Hi! I now figured out what was wrong with the original code in Firefox. Firefox removed the ObjectURL before the link was pressed, which made the download not work.

To correct this I took your original function and rewrote the clean-up part to 
    setTimeout(function() {
		window.URL.revokeObjectURL(url);
		a.remove();
	},1);

This makes the Javascript engine process all the pending events before it cleans up the ObjectURL

Have a download, and try it out in FF and Chrome so you can see it works! But I think the code is ready for merging now

**Pull request merged**